src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -867,6 +869,50 @@ private JsValue Raven_Min(JsValue self, JsValue[] args)
                 return args[0];
             }
 
+            private JsValue ArchiveAt(JsValue self, JsValue[] args)
+            {
+                if (args.Length != 2)
+                {
+                    throw new InvalidOperationException("archiveAt(doc, utcDateTimeString) must be called with two args");
+                }
+
+                if (args[0].IsNull() || args[0].IsUndefined())
+                    return args[0];
+                
+                if((args[0].IsObject() && args[0].AsObject() is BlittableObjectInstance) == false)
+                    throw new InvalidOperationException("archiveAt(doc, utcDateTimeString) must take document object as the first argument");
+
+                if (args[1].IsNull() || args[1].IsUndefined() || args[1].IsString() == false)
+                    throw new InvalidOperationException("archiveAt(doc, utcDateTimeString) must take string as the second argument");
+
+                if (DateTime.TryParse(args[1].ToString(), out DateTime _) == false)
```

Comment:
there is a method here `GetDateArg`, we should use it?

Reply:
done